src/ProjNavigator/sources_form.cpp

```diff
@@ -290,23 +294,55 @@ void SourcesForm::SlotRemoveFileSet() {
 }
 
 void SourcesForm::SlotRemoveFile() {
-  QTreeWidgetItem *item = m_treeSrcHierachy->currentItem();
-  if (item == nullptr) return;
+  // Get all selections
+  auto selectedItems = m_treeSrcHierachy->selectedItems();
+  // Use the r-clicked item for determining what item type we are working with
+  QTreeWidgetItem *refItem = m_treeSrcHierachy->currentItem();
+  auto itemType = refItem->data(0, Qt::WhatsThisPropertyRole);
+
+  // Bail on no selection
+  if (refItem == nullptr) return;
+  if (!selectedItems.count()) return;
+
+  // Track string names for confirmation dialog
+  QStringList files;
+  // Store filename/fileset pairs for delete option
+  QList<QPair<QString, QVariant>> selections;
+  for (auto item : selectedItems) {
+    auto strFileName = item->text(0);
+    auto fileSet = item->data(0, SetFileDataRole);
+    // Only accept files that match the r-clicked item type to avoid acting on
+    // un-related selections
+    if ((item->data(0, Qt::WhatsThisPropertyRole) == itemType) &&
+        !fileSet.isNull()) {
+      selections.append(qMakePair(strFileName, fileSet));
+      files.append(strFileName);
+    }
+  }
 
-  auto strFileName = item->text(0);
-  auto fileSet = item->data(0, SetFileDataRole);
-  if (!fileSet.isNull()) {
-    auto questionStr =
+  // Display delete confirmation message
+  if (files.count()) {
+    // Single File Message
+    QString questionStr =
         tr("Are you sure you want to remove %1 from the project? \n\nThe file "
            "will not be removed from the disk.")
-            .arg(strFileName);
+            .arg(files[0]);
+    if (files.count() > 1) {
+      // Multi File Message
+      questionStr = tr("Are you sure you want to remove the following files "
+                       "from the project?\n %1\n\nThey "
+                       "will not be removed from the disk.")
+                        .arg(files.join("\n "));
+    }
     if (QMessageBox::question(this, {}, questionStr) == QMessageBox::No) return;
-    m_projManager->setCurrentFileSet(fileSet.toString());
-    int ret = m_projManager->deleteFile(strFileName);
-    if (0 == ret) {
```

Comment:
@volodymyrkochyn When switching to a loop to delete multiple files, I decided to remove the logic of not updating/saving on ret != 0 because with multiple files, the next successful delete would update/save anyways. Prior to this change, i believe the same could be achieved by deleting a file (let's assume it somehow returned !0 thus not saving) and then deleting a different file that does return 0 as expected, in that case the project will still save even though the logic choose to not save on the first file failure so I believe the changed logic is still the same basically.

Let me know if you see any issue w/ this or if we really need to error out when the project manager fails to delete a file.

Reply:
I'm fine with it.